### PR TITLE
Add option to modify individual volunteering roles on application forms

### PR DESCRIPTION
**Pull request by vassyz**
This commit adds a "Change" link to the work history section of an application form. This link takes the support user to a new page where they can edit the details of a single volunteering roles. It reuses the candidate interface form for editing volunteering roles, but with a different controller and view.

 [Support: Edit "unpaid experience" functionality](https://trello.com/c/BNaBZzyV/827-support-edit-unpaid-experience-functionality)

### Adds a change button
Screenshot 2023-10-23 at 14 14 57

### That takes you to the edit volunteering role page:
Screenshot 2023-10-23 at 14 15 04

### CODE REVIEW

**Comment by vassyz** [comment]
[Support: Edit "unpaid experience" functionality](https://trello.com/c/BNaBZzyV/827-support-edit-unpaid-experience-functionality)

File: app/forms/shared/volunteering_role_form.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          currently_working: volunteering_role.currently_working.to_s,
+        )
+      end
+    end
+
+    def save(application_form)
+      return false unless valid?
+
+      application_form.application_volunteering_experiences.create!(map_attributes)
+      application_form.update!(volunteering_experience: true)
```
Comment (inulty-dfe) [inline app/forms/shared/volunteering_role_form.rb:76]: Maybe wrapping this in a transaction is safer

**Review by inulty-dfe — COMMENTED** [review]
Looks great, I don't understand why the `JobsController` was changed to `JobController` though. It's not a singular resource.

**Comment by vassyz** [comment]
> Looks great, I don't understand why the `JobsController` was changed to `JobController` though. It's not a singular resource.

I created the `JobsController` because as you mentioned, it's not a singular resource. But then I saw that the candidate one is called `RestructuredWorkHistory::JobController` and `RoleController` so I was just trying to keep things consistent.

File: app/forms/shared/volunteering_role_form.rb (hunk 1 above)
Comment (vassyz) [reply app/forms/shared/volunteering_role_form.rb]: That code got copied across. Added the transaction [here](https://github.com/DFE-Digital/apply-for-teacher-training/pull/8713/files#diff-4bf3f35ac586d93e16c4f2ac873ae892225ddfd2b50606aaa63d74016bf609f4R75).

**Comment by inulty-dfe** [comment]
> > Looks great, I don't understand why the `JobsController` was changed to `JobController` though. It's not a singular resource.
> 
> I created the `JobsController` because as you mentioned, it's not a singular resource. But then I saw that the candidate one is called `RestructuredWorkHistory::JobController` and `RoleController` so I was just trying to keep things consistent.

Now would be a fine time to correct  the error, but if you rather not do it now that's fine.

**Review by inulty-dfe — APPROVED** [review]

**Comment by vassyz** [comment]
> > > Looks great, I don't understand why the `JobsController` was changed to `JobController` though. It's not a singular resource.
> > 
> > 
> > I created the `JobsController` because as you mentioned, it's not a singular resource. But then I saw that the candidate one is called `RestructuredWorkHistory::JobController` and `RoleController` so I was just trying to keep things consistent.
> 
> Now would be a fine time to correct the error, but if you rather not do it now that's fine.

@inulty-dfe  https://github.com/DFE-Digital/apply-for-teacher-training/pull/8713/commits/eb66b722d5eb2027aa6f20365660f6ebae4a86ac

**Pull request merged**